modules/programs/autorandr.nix

```diff
@@ -114,6 +114,54 @@ let
           for the documentation of the transform matrix.
         '';
       };
+
+      dpi = mkOption {
+        type = types.nullOr types.int;
+        description = "Output DPI configuration.";
+        default = null;
+        example = 96;
+      };
+
+      scale = mkOption {
+        type = types.nullOr (types.submodule {
+          options = {
+            method = mkOption {
+              type = types.enum ["factor" "pixel" ];
+              description = "Output scaling method.";
+              default = "factor";
+              example = "pixel";
+            };
+
+            x = mkOption {
+              type = types.either types.float types.int;
+              description = "Horizontal scaling factor/pixels.";
+            };
+            y = mkOption {
+              type = types.either types.float types.int;
+              description = "Vertical scaling factor/pixels.";
+            };
+          };
+        });
+        description = ''
+          Output scale configuration.
+
+          Either configure by pixels or a scaling factor. When using pixel method the
+          <citerefentry>
+            <refentrytitle>xrandr</refentrytitle>
+            <manvolnum>1</manvolnum>
+          </citerefentry>
+          option --scale-from will be used; when using factor method the option --scale
```

Comment:
@rycee has a much stronger docbook fu than me, but I think `--scale-from` can also be wrapped in `<option></option>` tag.

Reply:
Had a look in the docbook documentation and apparently there is a special element for this, [parameter](https://tdg.docbook.org/tdg/5.0/parameter.html). If I understand the example in the link correctly this would be
```xml
<parameter class="command">--scale-from</parameter>
```
and
```xml
<parameter class="command">--scale</parameter>
```

We've used `<option>` before, though so I won't be picky about this. I think it's nice to use one of them, though, instead of leaving it bare.